src/components/masters/program/main-content/course-cards/email-settings/EmailSettingsModal.jsx

```diff
@@ -58,18 +59,31 @@ class EmailSettingsModal extends Component {
   handleOnClose = () => {
     this.setState({
       isSubmitting: false,
-      isFormChanged: false,
       error: null,
     });
     this.props.onClose();
   };
 
+  handleButtonState = () => {
+    const { isSubmitting, isSuccessful } = this.state;
+
+    if (isSubmitting) {
+      return 'pending';
+    } else if (isSuccessful) {
+      return 'complete';
+    }
+
+    return 'default';
+  }
+
   render() {
     const {
-      error, isFormChanged, hasEmailsEnabled, isSubmitting,
+      error, hasEmailsEnabled, isSubmitting,
     } = this.state;
     const { title, open } = this.props;
 
+    const buttonState = this.handleButtonState();
```

Comment:
nit: Is it necessary to define `buttonState` here? As opposed to calling `this.handleButtonState()` as the value of the `state` prop in `StatefulButton`?

Also, `getButtonState` may be a bit clearer of a name. `handleButtonState` leads me to think that function deals with some sort of event handler.

Reply:
Hi! I initially had state as this.handleButtonState() as a prop in state for the StatefulButton. However, proptypes doesn't seem to like that it is a function, which is why I handled it outside of the method. I can definitely change the name to getButtonState!